src/main/java/app/bpartners/api/service/subscription/SubscriptionService.java

```diff
@@ -69,6 +69,26 @@ public SubscriptionConsumptionLog addConsumption(
     return consumptionLogJpaRepository.save(subscriptionConsumptionLog);
   }
 
+  public SubscriptionConsumptionLog addConsumptionLog(
+      String apiKey, SubscriptionConsumptionLog subscriptionConsumptionLog) {
+    User user = userRepository.getUserByApiKey(apiKey);
```

Comment:
Do not integrate the authorization inside the service. Create an explicit Authorizer component, look at how in geo-jobs it is implemented.